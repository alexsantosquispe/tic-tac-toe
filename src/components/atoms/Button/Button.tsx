import type { ReactNode } from 'react';

interface ButtonProps {
  title: string;
  ariaLabel: string;
  onClick: () => void;
  icon?: ReactNode;
}

const Button = ({ title, ariaLabel, onClick, icon }: ButtonProps) => {
  return (
    <button
      className="flex items-center justify-center gap-2 rounded-xl bg-rose-600 px-5 py-3 text-2xl text-white uppercase shadow-md hover:cursor-pointer hover:bg-rose-500 dark:bg-rose-500 dark:hover:bg-rose-400"
      onClick={onClick}
      aria-label={ariaLabel}
    >
      {!!icon && icon}
      {title}
    </button>
  );
};

export default Button;
