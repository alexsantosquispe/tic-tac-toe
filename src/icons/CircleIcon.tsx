import type { SvgIconProps } from "../types";
import { twMerge } from "tailwind-merge";

export const CircleIcon = ({ className }: SvgIconProps) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="3"
      strokeLinecap="round"
      strokeLinejoin="round"
      className={twMerge("size-6", className)}
    >
      <circle cx="12" cy="12" r="10" />
    </svg>
  );
};
