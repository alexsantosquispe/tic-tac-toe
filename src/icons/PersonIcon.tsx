import { twMerge } from 'tailwind-merge';
import type { SvgIconProps } from '../types';

export const PersonIcon = ({ className }: SvgIconProps) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
      className={twMerge('size-6', className)}
    >
      <circle cx="12" cy="5" r="1" />
      <path d="m9 20 3-6 3 6" />
      <path d="m6 8 6 2 6-2" />
      <path d="M12 10v4" />
    </svg>
  );
};
