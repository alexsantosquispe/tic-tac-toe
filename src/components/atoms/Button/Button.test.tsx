import { fireEvent, render, screen } from '@testing-library/react';

import { ResetIcon } from '../../../icons/ResetIcon';
import Button from './Button';

describe('Button', () => {
  const onClickMock = jest.fn();
  const props = {
    title: 'Reset Button',
    ariaLabel: 'Reset Button',
    icon: <ResetIcon />,
    onClick: onClickMock
  };

  it('should render the component correctly', () => {
    const component = render(<Button {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('should call the onClick function when clicked', () => {
    render(<Button {...props} />);

    const button = screen.getByRole('button');

    fireEvent.click(button);

    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});
