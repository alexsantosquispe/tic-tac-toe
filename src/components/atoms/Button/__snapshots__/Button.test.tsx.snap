// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Button should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="Reset Button"
        class="flex items-center justify-center gap-2 rounded-xl bg-rose-600 px-5 py-3 text-3xl text-white uppercase shadow-md hover:cursor-pointer hover:bg-rose-500 dark:bg-rose-500 dark:hover:bg-rose-400"
      >
        <svg
          class="size-6"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="3"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 12a9 9 0 1 0 9-9 9.75 9.75 0 0 0-6.74 2.74L3 8"
          />
          <path
            d="M3 3v5h5"
          />
        </svg>
        Reset Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="Reset Button"
      class="flex items-center justify-center gap-2 rounded-xl bg-rose-600 px-5 py-3 text-3xl text-white uppercase shadow-md hover:cursor-pointer hover:bg-rose-500 dark:bg-rose-500 dark:hover:bg-rose-400"
    >
      <svg
        class="size-6"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="3"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 12a9 9 0 1 0 9-9 9.75 9.75 0 0 0-6.74 2.74L3 8"
        />
        <path
          d="M3 3v5h5"
        />
      </svg>
      Reset Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
